nodes/curve/ellipse_curve_mk2.py

```diff
@@ -0,0 +1,156 @@
+
+        list_links_remove = None
+        if self.mode == "AB":
+            list_links_remove = ["Eccentricity", "Focal Length",]
+        elif self.mode == "AE":
+            list_links_remove = ["Minor Radius", "Focal Length",]
+        else:  # AC
+            list_links_remove = ["Minor Radius", "Eccentricity",]
+
+        # remove existing link if sockets are switched:
+        for socket_name in list_links_remove:
+            all_sockets_name.remove(socket_name)
+            for l in self.inputs[socket_name].links:
+                self.id_data.links.remove(l)
+            self.inputs[socket_name].hide = True
+        self.inputs[ all_sockets_name[0] ].hide = False
+            
+        updateNode(self, context)
+
+    mode: EnumProperty(
+        name="Mode", items=mode_items,
+        description="Ellipse definition mode",
+        default="AB", update=update_sockets)
+
+    centering: EnumProperty(
+        name="Centering", items=centering_items,
+        description="Center the ellipse around F1, C or F2",
+        default=SvEllipse.CENTER,
+        update=update_sockets)
+
+    major_radius: FloatProperty(
+        name='Major Radius', description='Ellipse major radius (semiaxis)',
+        default=1.0, min=0.0, update=update_sockets)
+
+    minor_radius: FloatProperty(
+        name='Minor Radius', description='Ellipse minor radius (semiaxis)',
+        default=0.8, min=0.0, update=update_sockets)
+
+    eccentricity: FloatProperty(
+        name='Eccentricity', description='Ellipse eccentricity',
+        default=0.6, min=0.0, max=1.0, update=update_sockets)
+
+    focal_length: FloatProperty(
+        name='Focal Length', description='Ellipse focal length. Distance from ellipse’s center to it’s focal points',
+        default=0.6, min=0.0, update=update_sockets)
+
+    def draw_buttons(self, context, layout):
+        col = layout.column(align=True)
+        row = col.row(align=True)
+        row.prop(self, "mode", expand=True)
+        row = col.row(align=True)
+        row.prop(self, "centering", expand=True)
+
+    def sv_init(self, context):
+        self.width = 160
+        self.inputs.new('SvStringsSocket', "Major Radius").prop_name = "major_radius" # 0
+        self.inputs.new('SvStringsSocket', "Minor Radius").prop_name = "minor_radius" # 1
+        self.inputs.new("SvStringsSocket", "Eccentricity").prop_name = "eccentricity"
+        self.inputs.new("SvStringsSocket", "Focal Length").prop_name = "focal_length"
+        self.inputs.new('SvMatrixSocket', "Matrix") # 2
+
+        self.outputs.new('SvCurveSocket', "Ellipse")
+        self.outputs.new('SvVerticesSocket', "F1")
+        self.outputs.new('SvVerticesSocket', "F2")
+        
+        self.update_sockets(context)
+
+    def process(self):
+        outputs = self.outputs
+        # return if no outputs are connected
+        if not any(s.is_linked for s in outputs):
+            return
+
+        major_radius_s = self.inputs['Major Radius'].sv_get()
```

Comment:
`sv_get(deepcoopy=False)` for performance

Reply:
fixed